relayer/chains/evm/listener.go

```diff
@@ -71,13 +71,21 @@ func (p *Provider) Listener(ctx context.Context, lastProcessedTx relayertypes.La
 		case err := <-errChan:
 			if p.isConnectionError(err) {
 				p.log.Error("connection error", zap.Error(err))
-				client, err := p.client.Reconnect()
-				if err != nil {
-					p.log.Error("failed to reconnect", zap.Error(err))
-				} else {
-					p.log.Info("client reconnected")
-					p.client = client
+				clientReconnected := false
```

Comment:
This can be optimized. We don't need so many variables assignments rather error can be checked instead.

Reply:
Can you elaborate a bit ?? did you just mean pushing err variable within the loop or something else ?

Reply:
something like this?

```

if p.isConnectionError(err) {
	p.log.Error("connection error", zap.Error(err))
	for err != nil {
		p.log.Info("Reconnecting client")
		client, err := p.client.Reconnect()
		if err == nil {
			p.log.Info("client reconnected")
			p.client = client
		} else {
			p.log.Error("failed to re-connect", zap.Error(err))
			time.Sleep(ClientReconnectDelay)
		}
	}
}
```

Reply:
updated the code as per feedback.